suprb/optimizer/rule/ns/novelty_search_type.py

```diff
@@ -0,0 +1,62 @@
+from abc import ABCMeta, abstractmethod
+
+import numpy as np
+from suprb.rule import Rule
+from suprb.base import BaseComponent
+
+
+class NoveltySearchType(BaseComponent, metaclass=ABCMeta):
+    """Abstract class of Novelty Search Type"""
+
+    def filter_rules(self, rules: list[Rule]):
```

Comment:
as this is not an abstractmethod you might as well implement the most common functionality here rather than in the inheriting classes